include/nbl/video/utilities/IUtilities.h

```diff
@@ -153,8 +162,29 @@ class IUtilities : public core::IReferenceCounted
         //! Remember to ensure a memory dependency between the command recorded here and any users (so fence wait, semaphore when submitting, pipeline barrier or event)
         inline core::smart_refctd_ptr<IGPUImage> createFilledDeviceLocalGPUImageOnDedMem(IGPUCommandBuffer* cmdbuf, IGPUImage::SCreationParams&& params, const IGPUImage* srcImage, uint32_t regionCount, const IGPUImage::SImageCopy* pRegions)
         {
-            // Todo(achal): srcImage's format should support VK_FORMAT_FEATURE_TRANSFER_SRC_BIT,
-            // dstImage's format should support VK_FORMAT_FEATURE_TRANSFER_DST_BIT
+            // This API check is temporary (or not?) since getFormatProperties is not
+            // yet implemented on OpenGL
+            if (srcImage->getAPIType() == EAT_VULKAN)
+            {
+                const auto* physicalDevice = srcImage->getOriginDevice()->getPhysicalDevice();
```

Comment:
IUtilities has a `m_logicalDevice` which you can use instead of `getOriginDevice()`

Reply:
Resolved